### PR TITLE
Hotfix: Cell location calculation, mat_map verifications

### LINKED ISSUE #6: Material map n too low for desired layout
For the `mat_map` class, specifying an `n` that is not large enough to ensure every cell has only one material does not throw an error, only returns an empty `mat_lib.array`

For example, `n=2` with the layout `1 1 1 1   1 2 2 1    1 2 2 1   1 1 1 1` would define four squares with two different materials.

### LINKED ISSUE #21: list out of index when retrieving material ID at a specific cell
When retrieving material ID for a cell, it shows the error in the title. Here's how it was used:
```
class SAAF(object):
    def __init__(self, mat_cls, mesh_cls, prob_dict):
        self._mesh=mesh_cls
```

When using asking for ID in a cell:

```
def some_func(self):
    for cell in self._mesh.cells():
        global_idx,mid = cell.global_idx(),cell.get('id')
```

### PR DESCRIPTION
**Pull request by jsrehak**
This contains two hotfixes:
1. Adds verification to initialization of a material map (`mat_map`), the number of mesh cells (in one direction) given as `n` must be a _multiple_ of the size of the layout. This is required because all cells are required to have just *one* material. Also verifies that the mesh cell number is larger than the layout and larger than 0. Resolves #21 and Fixes #6.

2. Fixes a bug in `cell.get()` that reversed calculation of `x` and `y`. Note that cell index, from `cell.index()` is given in a row-column format. That is `(0,5)` is the cell in row 0, column 5.

Also contains the new UOX input file.

### CODE REVIEW

File: mesh.py [hunk 1]
```diff
@@ -167,8 +167,8 @@ def __init__(self, index, mesh_params, mat_map=None):
     # MATERIAL PROPERTIES  ==============================================
     def get(self, prop):
         try:
-            x = self._length*(self._index[0] + 0.5)
-            y = self._length*(self._index[1] + 0.5)
+            x = self._length*(self._index[1] + 0.5)
+            y = self._length*(self._index[0] + 0.5)
```
Comment (marissachanlatte) [inline mesh.py:172]: Just to check, why did this index flip?

**Review by marissachanlatte — COMMENTED** [review]
Looks good, just one part I was unclear about.

**Comment by jsrehak** [comment]
When defining the cells, the `cell.index()` is stored as `(i,j)` where `i` is row and `j` is column. So `index[0]` is the row number and will therefore set the `y` coordinate (same reasoning for `x`)

File: mesh.py (hunk 1 above)
Comment (marissachanlatte) [reply mesh.py:172]: ahhh makes sense. thanks!

**Pull request merged**